PR_REVIEW_CHECKLIST.md

```diff
@@ -0,0 +1,98 @@
+3. Should we add regression testing for code introduced in this PR?
+    - If the PR introduces **new** pages, features, etc. then we **should** have new regression tests written. Please note this in your PR so that Applause can make sure it gets added.
+    - If the PR fixes a bug, Applause will check to make sure the existing regression testing steps already cover the workflow fixed by the bug.
+
+# Reviewing the code
+
+## Good code patterns to require
+
+1. Check that functions have comments when appropriate.
+    - If the function has params or returns something, use JSDocs syntax to describe them.
+        - Indicate the param name(s), data type passed and / or returned, and purpose if not immediately obvious.
+    - Obvious functions (with no params / return value) should not have comments.
+    - **In short: _Add comments & docs often, only when useful._**
+2. All copy / text shown to users in the product should be stored in the `src/languages/*` files for localization.
+3. Platform-specific files should follow the proper naming convention (see [Platform-Specific File Extensions](https://github.com/expensify/app#platform-specific-file-extensions))
+    - Example: `MyComponent/index.android.js`
+
+## Code patterns to check for improvement
+
+1. Is there appropriate error handling?
+2. Reuse...
+3. Are there any changes that could've been applied to one component that were applied to every usage of the component instead?
+    - Example: https://github.com/Expensify/App/pull/3129#pullrequestreview-668271068
+4. Ensure fixes for just android or ios aren't put in a `index.native.js`. Instead they should be in a specific `index.ios.js` or `index.android.js` file.
+    - Put another way, ensure platform-specific solutions are not applied cross-platform.
+5. Look for usages of react fragment wrappings and confirm they are necessary.
+    - Also look for any unnecessary `View`s and think if they should be replaced with React fragments.
+6. Were any new styles added? Can we reuse existing styles instead? Can we generalize the new style?
+
+## Code patterns to avoid
+
+1. Make sure any `setTimeout`s are absolutely necessary, not just a workaround for a situation that isn’t fully understood.
+2. Ensure Onyx is used appropriately. ([docs](https://github.com/expensify/react-native-onyx#merging-data))
+3. Don't mention the `isSmallScreenWidth` prop type explicitly in any component, use `...windowDimensionsPropTypes` instead.
+
+## Considerations around code reuse
+
+While trying to keep our code DRY (don't repeat yourself), you may find you're walking a thin line between refactoring & reusing a component vs. creating a brand new component. Here are some guidelines we use - they're not hard and fast rules, so please feel free to post in #expensify-open-source to gather peer opinions in order to make the best decisions.
```

Comment:
NAB, idioms like "walk a thin line" add flavor but are less direct than just telling people straight up what DRY means, how to identify opportunities for it, etc

Reply:
Agreed, I thought we agreed that this topic (specialization vs reuse) wasn't 100% clear, meaning it won't be easy to give super direct instruction. Would this be a better intro:

> While trying to keep our code DRY (don't repeat yourself), you may consider refactoring & reusing a component vs. creating a brand new specialized component. Here are some guidelines we use - they're not hard and fast rules, so please feel free to post in #expensify-open-source to gather peer opinions in order to make the best decisions.